recipes/assimp/5.x/conanfile.py

```diff
@@ -240,20 +258,52 @@ def _patch_sources(self):
             ('SET(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} /Zi")', ""),
             ('SET(CMAKE_SHARED_LINKER_FLAGS_RELEASE "${CMAKE_SHARED_LINKER_FLAGS_RELEASE} /DEBUG:FULL /PDBALTPATH:%_PDB% /OPT:REF /OPT:ICF")', ""),
-            ("/WX", "")
+            ("/WX", ""),
+            ("-Werror", ""),
         ]
 
         for before, after in replace_mapping:
-            replace_in_file(self, os.path.join(
-                self.source_folder, "CMakeLists.txt"), before, after, strict=False)
-        # Take care to not use these vendored libs
-        vendors = ["poly2tri", "rapidjson", "utf8cpp", "zip", "unzip", "stb", "zlib", "clipper"]
-        if Version(self.version) < "5.1.0":
-            vendors.append("irrXML")
-        else:
-            vendors.extend(["pugixml", "draco", "openddlparser"])
-        for vendor in vendors:
-            rmdir(self, os.path.join(self.source_folder, "contrib", vendor))
+            replace_in_file(self, os.path.join(self.source_folder, "CMakeLists.txt"), before, after, strict=False)
+            replace_in_file(self, os.path.join(self.source_folder, "code", "CMakeLists.txt"), before, after, strict=False)
+
+        # Take care to not use vendored libs
+        allow_vendored = ["Open3DGC"]
+        for contrib_dir in self.source_path.joinpath("contrib").iterdir():
+            if contrib_dir.is_dir() and contrib_dir.name not in allow_vendored:
+                rmdir(self, contrib_dir)
+
+        # Do not build vendored libs
+        code_cmakelists = self.source_path.joinpath("code", "CMakeLists.txt")
+        content = code_cmakelists.read_text(encoding="utf-8")
+        for vendored_lib in [
+            "unzip_compile",
+            "Poly2Tri",
+            "Clipper",
+            "openddl_parser",
+            # "open3dgc",
+            "ziplib",
+            "Pugixml",
+            "stb",
+        ]:
+            content = content.replace("${%s_SRCS}" % vendored_lib, "")
+        code_cmakelists.write_text(content, encoding="utf-8")
+
+        # Make vendored headers redirect to external ones
+        for contrib_header, include in [
+            (os.path.join("clipper", "clipper.hpp"), "polyclipping/clipper.hpp"),
+            (os.path.join("poly2tri", "poly2tri", "poly2tri.h"), "poly2tri/poly2tri.h"),
+            (os.path.join("stb", "stb_image.h"), "stb_image.h"),
+            (os.path.join("utf8cpp", "source", "utf8.h"), "utf8.h"),
+            (os.path.join("zip", "src", "zip.h"), "zip/zip.h"),
+        ]:
+            save(self, os.path.join(self.source_folder, "contrib", contrib_header),
+                 f"#include <{include}>\n")
+
+        # misc
+        replace_in_file(self, os.path.join(self.source_folder, "CMakeLists.txt"),
+                        "use_pkgconfig(UNZIP minizip)", "set(UNZIP_FOUND TRUE)")
+        replace_in_file(self, os.path.join(self.source_folder, "CMakeLists.txt"),
+                        "INSTALL( TARGETS zlib", "message(TRACE #", strict=False)
```

Comment:
Never use `strict=False` please. It can hide patch issues.

Reply:
I agree. Fixed.